_episodes/11-dask-configuration.md

```diff
@@ -0,0 +1,390 @@
+
+We use a text editor called ``nano`` to edit the configuration file:
+
+~~~bash
+  nano ~/.esmvaltool/dask.yml
+~~~
+
+Any other editor can be used, e.g. many systems have ``vi`` available.
+
+This file contains the settings for:
+
+- Starting a new cluster of Dask workers
+- Or alternatively: connecting to an existing cluster of Dask workers
+
+Add the following content to the file ``~/.esmvaltool/dask.yml``:
+
+```yaml
+cluster:
+  type: distributed.LocalCluster
+  n_workers: 1
+  threads_per_worker: 2
+  memory_limit: 4GiB
+```
+
+This tells ESMValCore to start a new cluster of one worker, that can use 2
+gigabytes (GiB) of memory and run computations using 2 threads. For a more
+extensive description of the available arguments and their values, see
+[``distributed.LocalCluster``](https://distributed.dask.org/
+en/stable/api.html#distributed.LocalCluster).
+
+To see this configuration in action, run we will run a version of
+[recipe_easy_ipcc.yml](https://docs.esmvaltool.org/
+en/latest/recipes/recipe_examples.html) with just two datasets.
+This recipe takes a few minutes to run, once you have the data available.
+Download the recipe [here](../files/recipe_easy_ipcc_short.yml) and run it
+with the command:
+
+~~~bash
+  esmvaltool run recipe_easy_ipcc_short.yml
+~~~
+
+After finding and downloading all the required input files, this will start
+the Dask scheduler and workers required for processing the data. A message that
+looks like this will appear on the screen:
+
+```
+2024-05-29 12:52:38,858 UTC [107445] INFO    Dask dashboard: http://127.0.0.1:8787/status
+```
+
+Open the Dashboard link in a browser to see the Dask Dashboard website.
+When the recipe has finished running, the Dashboard website will stop working.
+The top left panel shows the memory use of each of the workers, the panel on the
+right shows one row for each thread that is doing work, and the panel at the
+bottom shows the progress of all work that the scheduler currently has been
+asked to do.
+
+> ## Explore what happens if workers do not have enough memory
+>
+> Reduce the amount of memory that the workers are allowed to use to 2GiB and
+> run the recipe again. Watch what happens.
+>
+>> ## Solution
+>>
+>> We use `memory_limit` entry in the `~/.esmvaltool/dask.yml` file to set the
+>> amount of memory allowed to 2GiB:
+>>```yaml
+>> cluster:
+>>    type: distributed.LocalCluster
+>>    n_workers: 1
+>>    threads_per_worker: 2
+>>    memory_limit: 2GiB
+>>```
+>> Note that the bars representing the memory use turn
+>> orange as the worker reaches the maximum amount of memory it is
+>> allowed to use and it starts 'spilling' (writing data temporarily) to disk.
+>> The red blocks in the top right panel represent time spent reading/writing
+>> to disk. While 2 GiB per worker may be enough in other cases, it is
+>> apparently not enough for this recipe.
+>>
+> {: .solution}
+{: .challenge}
+
+
+> ## Tune the configuration to your own computer
+>
```

Comment:
Perhaps a not here to tell users to be mindful when on shared servers such as JASMIN or Levante. What can they do and what they should not? For instance, the sci6 node on JASMIN has ~1000GB memory which I found out using the memcopy file in your instructions. I don't know if the sys admin has guidelines for how much a user can use but if you have thoughts/general suggestions for this, I think it might be useful.

Reply:
I added a note to please respect other users on line 180